doc(kic): upgrade to KIC v3.5 (#2259)
Kong/developer.konghq.com: app/kubernetes-ingress-controller/faq/upgrading-ingress-controller.md

1. Install the Experimental Channel of Gateway API v1.1:

   ```sh

>    kubectl apply --force=true -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.1.0/experimental-install.yaml

Author: Good point! Let me see how to add it.

> Starting in {{ site.kic_product_name }} 3.4, {{ site.kic_product_name }} supports Gateway API version 1.2.
> There is a breaking change in Gateway API 1.2 to remove the `v1alpha2` version of `GRPCRoute` and `ReferenceGrant`.

Reviewer: ```suggestion
Starting in {{ site.kic_product_name }} 3.4, {{ site.kic_product_name }} supports Gateway API version 1.2.
Gateway API 1.2 introduces a breaking change by removing the `v1alpha2` versions of `GRPCRoute` and `ReferenceGrant.`
```

> If you have been using Gateway API v1.1 and {{ site.kic_product_name }} 3.2 and above, and there are no `GRPCRoute` and `ReferenceGrant` resources stored in `v1alpha2` version, you can directly upgrade Gateway API from v1.1 to v1.2.

Reviewer: ```suggestion
If you are using Gateway API v1.1 with {{ site.kic_product_name }} 3.2 or later, and you do not have any `GRPCRoute` or `ReferenceGrant` resources in the `v1alpha2` version, you can upgrade directly to Gateway API v1.2.
```

> 3. Update all your `GRPCRoute` manifests to use `v1` instead of `v1alpha2` and your `ReferenceGrant` manifests to use `v1beta1` instead of `v1alpha2`. This can be done by following the [upgrade guide from Gateway API](https://gateway-api.sigs.k8s.io/guides/?h=v1.2#v12-upgrade-notes).

Reviewer: ```suggestion
3. Update all `GRPCRoute` manifests to use `v1` instead of `v1alpha2`, and all `ReferenceGrant` manifests to use `v1beta1` instead of `v1alpha2`. You can follow the [Gateway API upgrade guide](https://gateway-api.sigs.k8s.io/guides/?h=v1.2#v12-upgrade-notes) for detailed steps.
```

4. Install the standard or experimental channel of Gateway API 1.2.

### {{ site.kic_product_name }} 3.5

> Starting in {{ site.kic_product_name }} 3.5, several Kong-specific resources have been deprecated in favor of their Gateway API equivalents, and CRDs are now managed separately.

Reviewer: ```suggestion
Starting from {{ site.kic_product_name }} 3.5, several Kong-specific resources have been deprecated in favor of their Gateway API equivalents, and CRDs are now managed separately.
```

#### Resource Deprecations

{{ site.kic_product_name }} 3.5 deprecates several Kong-specific ingress resources. While these resources continue to function, you will see deprecation warnings when using them. The recommended migration paths are:

* `KongIngress` → `KongUpstreamPolicy`
* `TCPIngress` → `TCPRoute` (from Gateway API)
* `UDPIngress` → `UDPRoute` (from Gateway API)

These deprecated resources will be removed in a future version. We recommend migrating to the Gateway API equivalents as soon as possible.

#### Gateway API Support

> Starting in {{ site.kic_product_name }} 3.5, {{ site.kic_product_name }} supports Gateway API version 1.3.

Reviewer: ```suggestion
Starting from {{ site.kic_product_name }} 3.5, {{ site.kic_product_name }} supports Gateway API version 1.3.
```

To upgrade Gateway API from 1.2 to 1.3:

1. Ensure you are using {{ site.kic_product_name }} version 3.5 or later.
2. Install Gateway API 1.3 CRDs:

   ```bash
   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.3.0/standard-install.yaml
   ```

   For experimental features like CORS filters, install the experimental channel:

   ```bash
   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.3.0/experimental-install.yaml
   ```

3. [Upgrade to {{ site.kic_product_name }} v3.5](#upgrade-kong-ingress-controller).

#### New Features

> {{ site.kic_product_name }} 3.5 introduces several new features:
> 
> * **Combined HTTP Routes (GA):** Now generally available, this feature allows consolidation of multiple `HTTPRoute` resources into a single {{site.base_gateway}} service. Enable with the `--combined-services-from-different-httproutes` flag.
> * **Connection Draining:** Ensures graceful handling of client connections to terminating pods. Enable with the `--enable-drain-support` flag.

Reviewer: ```suggestion
The {{ site.kic_product_name }} 3.5 release introduces several new features:

* **Combined HTTP Routes (GA):** Now generally available, this feature allows you to consolidate multiple `HTTPRoute` resources into a single {{ site.base_gateway }} service. Enable with the `--combined-services-from-different-httproutes` flag.
* **Connection Draining:** Ensures graceful handling of client connections when pods terminate. Enable with the `--enable-drain-support` flag.
```

1. **Remove or rename outdated CLI arguments and `CONTROLLER_*` environment variables.**

>     Version 3.0 removes or renames several flags that were previously deprecated, were removed due to other changes, or were left over for compatibility after their functionality was removed.
> 
>     The CLI argument versions of flags are listed below. You must also change the equivalent `CONTROLLER_` (for example, `CONTROLLER_SYNC_RATE_LIMIT` for `--sync-rate-limit`) environment variable if you use those.

Reviewer: ```suggestion
    Version 3.0 removes or renames several flags that were previously deprecated, removed due to other changes, or retained temporarily for compatibility after their functionality was removed.

    The CLI argument versions of these flags are listed below. If you use the equivalent `CONTROLLER_` environment variables (for example, `CONTROLLER_SYNC_RATE_LIMIT` for `--sync-rate-limit`), you must update those as well.
```
